backend/Models/Groups/group-model.js

```diff
@@ -0,0 +1,434 @@
+            [
+                description,
+                username,
+                group_id,
+
+            ],function(error, results) {
+                if (error) {
+                  reject(error);
+                  return;
+                }
+                resolve(results);
+              }
+            );
+          });
+}
+/**
+ *
+ * @param {Object} param0
+ * @param {String} param0.username
+ * @param {Number} param0.group_id
+ * @return {Promise}
+ */
+function deleteGroup({
+    username,
+    group_id : group_id,
+}) {
+   return new Promise((resolve,reject) => {
+    pool.query(`DELETE FROM groups WHERE id = ? AND
+    (SELECT COUNT(username) FROM UserGroups WHERE (username=? AND admin=1 AND group_id = ?))`,
+    [
+        group_id,
+        username,
+        group_id,
+
+    ],function(error, results) {
+        if (error) {
+          reject(error);
+          return;
+        }
+        resolve(results);
+      }
+    );
+  });
+}
+/**
+ *
+ * @param {Object} param0
+ * @param {String} param0.username
+ * @param {String} param0.group_id
+ * @param {String} param0.admin  
+ * @param {Number} param0.Userid
+ * @return {Promise}
+ */
+
+
+function addUserToGroup({
+    username,
+    group_id: group_id,
+    admin: admin,
+    usernameToAdd : usernameToAdd
+}){
+    return new Promise(function(resolve, reject) {
+      pool.query(`INSERT INTO UserGroups(username,group_id,admin) VALUES (?,?,?) WHERE
+      (SELECT COUNT(username) FROM UserGroups WHERE (username=? AND admin=1) `,
+    [
+      usernameToAdd,
+      group_id,
+      admin,
+      username,
+    ],
+        function(error, results) {
+          if (error) {
+            reject(error);
+            return;
+          }
+          resolve(results);
+        }
+      );
+    });
+  }
+
+/**
+ *
+ * @param {Object} param0
+ * @param {String} param0.username
+ * @param {String} param0.group_id
+ * @param {Number} param0.usernameToChange
+ * @return {Promise}
+ */
+
+
+function makeUserAdmin({
+    username,
+    group_id: group_id,
+    admin: admin,
+    usernameToChange : usernameToChange
+}){
+    return new Promise(function(resolve, reject) {
+      pool.query(`UPDATE UserGroups SET admin = 1 WHERE username=? AND group_id = ?
+      (SELECT COUNT(username) FROM UserGroups WHERE (username=? AND admin=1) `,
+    [
+      usernameToChange,
+      group_id,
+      username,
+    ],
+        function(error, results) {
+          if (error) {
+            reject(error);
+            return;
+          }
+          resolve(results);
+        }
+      );
+    });
+  }
+
+  /**
+  * @param {Object} param0
+  * @param {String} param0.username
+ ` *@param {String} param0.name
+  * @param {String} param0.description
+  * @param {Number} param0.confidential
+  * @param {String} param0.branch
+  * @param {String} param0.semester
+  * @return {Promise}
+  */
+ function customGroup(
```

Comment:
Also create an endpoint to copy an existing non confidential group.

Reply:
Done